src/Microsoft.Azure.Functions.Worker.Extensions.OpenApi/OpenApiHttpTriggerContext.cs

```diff
@@ -235,27 +236,43 @@ private string GetRuntimePath(string functionAppDirectory, bool appendBin)
         // This method relies on the dependency manifest file to find the function app runtime dll file.
         // It can be either <your_runtime>.deps.json or function.deps.json. In most cases, at least the
         // function.deps.json should exist, but in case no manifest exists, it will throw the exception.
+        // In case there are multiple .deps.json files, the root project will be picked, based on the
+        // dependencies mentioned in the .deps.json files.
         private async Task<string> GetRuntimeFilenameAsync(string functionAppDirectory)
         {
             var files = Directory.GetFiles(functionAppDirectory, "*.deps.json", SearchOption.AllDirectories);
-            var file = files.FirstOrDefault();
-            if (file.IsNullOrWhiteSpace())
+            if (!files.Any())
             {
                 throw new InvalidOperationException("Invalid function app directory");
             }
 
+            var dependencyManifests = new List<DependencyManifest>();
+            foreach (var file in files) {
+                dependencyManifests.Add(await GetDependencyManifestAsync(file));
+            }
```

Comment:
Can we use the opening brace at the next line of the `foreach`?

Reply:
Thanks, totally missed that one!

Reply:
Thanks for the update!